core-api/src/main/java/com/optimizely/ab/config/audience/match/StringAttributeCondition.java

```diff
@@ -0,0 +1,63 @@
+/**
+ *
+ *    Copyright 2019, Optimizely and contributors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab.config.audience.match;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+/**
+ * Matches content of {@link String} against any {@link CharSequence} input.
+ */
+public class StringAttributeCondition extends AbstractAttributeCondition<String, CharSequence> {
```

Comment:
It's surprising that a class with this name is only used for _exact_ string conditions and not for substring conditions.